### PR TITLE
Replace resize_dim with set_sizes_and_strides in THTensor_(squeeze)

### PR DESCRIPTION
**Pull request by HuitongQiu**
Summary: Replace resize_dim() with set_sizes_and_strides() in   THTensor_(squeeze) in aten/src/TH/generic/THTensor.cpp

Differential Revision: D14471066

### CODE REVIEW

File: aten/src/TH/generic/THTensor.cpp [hunk 1]
```diff
@@ -438,20 +438,24 @@ void THTensor_(squeeze)(THTensor *self, THTensor *src)
...
-      if(d != ndim)
-      {
-        self->set_size(ndim, src->size(d));
-        self->set_stride(ndim, src->stride(d));
+      if (d != ndim) {
+        newSize.push_back(src->size(d));
+        newStride.push_back(src->stride(d));
+      } else {
+        newSize.push_back(self->size(ndim));
+        newStride.push_back(self->stride(ndim));
```
Comment (ezyang) [inline aten/src/TH/generic/THTensor.cpp]: This conditional is very confusing. If `d != ndim`, you use d. But if `d == ndim`, you use ndim. But it's equivalent to use d. You don't need the if statement.

**Review by ezyang — CHANGES_REQUESTED** [review]
I recommend reading the docs to see what `torch.squeeze` does; it will make it clear how you should write this loop.

File: aten/src/TH/generic/THTensor.cpp (hunk 1 above)
Comment (HuitongQiu) [reply aten/src/TH/generic/THTensor.cpp]: I think if statement is still needed. If `d != ndim`, the size/stride of `src` is used, while if `d == ndim`, the size/stride of `self` is used.

**Review by ezyang — APPROVED** [review]
Much better!

**Pull request closed without merging**